addons/pos_stripe/models/pos_payment_method.py

```diff
@@ -0,0 +1,105 @@
+        if not self.sudo()._get_stripe_secret_key():
+            raise ValidationError(_('Complete the Stripe onboarding.'))
+
+        for payment_method in self:
+            if not payment_method.stripe_serial_number:
+                continue
+            existing_payment_method = self.search([('id', '!=', payment_method.id),
+                                                   ('stripe_serial_number', '=', payment_method.stripe_serial_number)],
+                                                  limit=1)
+            if existing_payment_method:
+                raise ValidationError(_('Terminal %s is already used on payment method %s.',\
+                     payment_method.stripe_serial_number, existing_payment_method.display_name))
+
+
+    @api.model
+    def _get_stripe_secret_key(self):
+        stripe_secret_key = self.env['payment.acquirer'].search([('provider', '=', 'stripe')], limit=1).stripe_secret_key
+
+        if not stripe_secret_key:
+            raise ValidationError(_('Complete the Stripe onboarding.'))
+
+        return stripe_secret_key
+
+    @api.model
+    def stripe_connection_token(self):
+        if not self.env.user.has_group('point_of_sale.group_pos_user'):
+            raise AccessError(_("Do not have access to fetch token from Stripe"))
+
+        endpoint = 'https://api.stripe.com/v1/terminal/connection_tokens'
+
+        try:
+            req = requests.post(endpoint, auth=(self.sudo()._get_stripe_secret_key(), ''), timeout=TIMEOUT)
+        except requests.exceptions.RequestException:
+            raise UserError(_("There are some issues between us and Stripe, try again later."))
+
+        if req.status_code is not requests.codes.ok:
+            raise UserError(_("There are some issues to contact Stripe."))
+
+        return req.json()
+
+    def stripe_payment_intent(self, amount):
+        if not self.env.user.has_group('point_of_sale.group_pos_user'):
+            raise AccessError(_("Do not have access to fetch token from Stripe"))
+
+        # For Terminal payments, the 'payment_method_types' parameter must include
+        # 'card_present' and the 'capture_method' must be set to 'manual'
+        endpoint = 'https://api.stripe.com/v1/payment_intents'
+        currency = self.journal_id.currency_id or self.company_id.currency_id
+
+        try:
+            data = werkzeug.urls.url_encode({
+                "currency": currency.name,
+                "amount": int(amount/currency.rounding),
+                "payment_method_types[]": "card_present",
+                "capture_method": "manual",
+            })
+            req = requests.post(endpoint, data=data, auth=(self.sudo()._get_stripe_secret_key(), ''), timeout=TIMEOUT)
+        except requests.exceptions.RequestException:
+            raise UserError(_("There are some issues between us and Stripe, try again later."))
+
+        if req.status_code is not requests.codes.ok:
+            raise UserError(_("There are some issues to contact Stripe."))
```

Comment:
nit: you might want to name your `Response` variables `resp` or `result` rather than `req` (it's not a request)
And then you could also use the shorthand `resp.ok`[^1] to test that the status_code is < 400, which is probably enough for your purpose.

As a best practice you should also log the problem somehow. It will be hard to investigate otherwise.

An example combining the various things:

```py
    try:
        resp = requests.post(...)
    except requests.exceptions.RequestException:
        # log the exception trace
        _logger.exception("Failed to call payment_intents endpoint")
        raise UserError(_("..."))

    if resp.ok:
        return resp.json()

    _logger.error("Unexpected payment_intents response: %s", resp.status_code)
    raise UserError(_("..."))
```

[^1]: https://requests.readthedocs.io/en/latest/api/#requests.Response.ok